src/lxd_type.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Types used by Lxd class."""
+
+from __future__ import annotations
+
+from typing import TypedDict
+
+# The keys are not valid identifiers, hence this is defined with function-based syntax.
+ResourceProfileConfig = TypedDict(
+    "ResourceProfileConfig", {"limits.cpu": str, "limits.memory": str}
+)
+ResourceProfileConfig.__doc__ = "Represent LXD profile configuration."
+
+
+class ResourceProfileDevicesRoot(TypedDict):
+    """Represents LXD device profile."""
+
+    path: str
+    pool: str
+    type: str
+    size: str
```

Comment:
I think it might be helpful if the attributes could be explained with docstrings, for instance `size: str`: `size in integer with unit. E.g. 10G`